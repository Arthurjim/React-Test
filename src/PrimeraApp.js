import React from "react";
import PropTypes from 'prop-types'
const PrimeraApp = ({saludo, subtitulo="SUBTITULO"}) => {
   
   // console.log(props.saludo);
   return (
      <>
         <h1>{saludo}</h1>
         {/* <pre>{ JSON.stringify(saludo) }</pre> */}
         <p>{subtitulo}</p>
      </>
   );
};

PrimeraApp.propTypes={
   saludo: PropTypes.string.isRequired
}


//props por defecto
PrimeraApp.defaultProps={
   // subtitulo: 'Soy un Subtitlo'
}

export default PrimeraApp;
