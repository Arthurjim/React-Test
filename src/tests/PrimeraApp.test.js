import React from 'react'
import { shallow } from 'enzyme';
import PrimeraApp from "../PrimeraApp";
import '@testing-library/jest-dom'
describe('Pruebas en <PrimeraApp />',()=>{
    // test('Debe de mostrar el mensaje "Hola soy goku',()=>{

    //     const saludo = 'Hola, soy Goku';

    //     // const wrapper = render( <PrimeraApp saludo={saludo} />)
    //     const {getByText} = render( <PrimeraApp saludo={saludo} />)
        
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('Debe de mostrar <PrimerApp/> correctamente', () => {
        //shallow nos ayuda a simluar clicks o el mismo document
        const saludo = "Hola soy tu papi"
        const wrapper = shallow( <PrimeraApp saludo={saludo} />)

        // lo que hace es tomar una "captura" de lo que se tiene en el componente 
        //y lo guarda en la carpeta __snapsshot_ manda error si los dos archivos no coinciden
        expect( wrapper ).toMatchSnapshot();

    })
    
    test('Debe de mostrar el subitulo enviado por props',()=>{
        const saludo = "Hola soy tu papi"
        const subtitulo = "Soy un subtitulo"

        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo} 
                subtitulo={subtitulo}
            />)
        const textoParrafo = wrapper.find('p').text()
        expect(textoParrafo).toBe(subtitulo)
    })

})