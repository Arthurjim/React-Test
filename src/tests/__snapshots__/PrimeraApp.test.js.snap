// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe de mostrar <PrimerApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy tu papi
  </h1>
  <p>
    SUBTITULO
  </p>
</Fragment>
`;
